src/o365/spo/commands/site/site-rename.ts

```diff
@@ -0,0 +1,285 @@
+import commands from '../../commands';
+import request from '../../../../request';
+import SpoCommand from '../../../base/SpoCommand';
+import { CommandOption, CommandValidate, CommandCancel } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import { FormDigestInfo} from '../../spo';
+
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  newSiteUrl: string;
+  newSiteTitle?: string;
+  suppressMarketplaceAppCheck?: boolean;
+  suppressWorkflow2013Check?: boolean;
+  wait?:boolean;
+}
+
+class SpoSiteRenameCommand extends SpoCommand {
+  private context?: FormDigestInfo;
+  private timeout?: NodeJS.Timer;
+  private operationData: any;
+  
+  public get name(): string {
+    return commands.SITE_RENAME;
+  }
+
+  public get description(): string {
+    return 'Renames the URL and title of a site collection';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.newSiteTitle = args.options.newSiteTitle ? true : false;
+    telemetryProps.suppressMarketplaceAppCheck = args.options.suppressMarketplaceAppCheck;
+    telemetryProps.suppressWorkflow2013Check = args.options.suppressWorkflow2013Check;
+    telemetryProps.wait = args.options.wait;
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    let spoAdminUrl: string = "";
+    let options = args.options;
+
+
+    this
+      .getSpoAdminUrl(cmd, this.debug)
+      .then((_spoAdminUrl: string): Promise<FormDigestInfo> => {
+        spoAdminUrl = _spoAdminUrl;
+
+        return this.getRequestDigest(spoAdminUrl);
+      })
+      .then((res: FormDigestInfo): Promise<string> => {
```

Comment:
The return value of this call is not a `string` but the `operationData` information